http-server/handlers/handlers.go

```diff
@@ -0,0 +1,489 @@
+			date = time.Now().Format("20060102")
+		case task.Repeat != "" && task.Date != "":
+			dataTime, err := time.Parse("20060102", task.Date)
+			if err != nil {
+				msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"Parse to date error"}, false)
+				w.WriteHeader(errInt)
+				w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+				w.Write(msg)
+				return
+			}
+			if dataTime.After(time.Now()) {
+				date = task.Date
+			} else {
+				date, err = api.NextDate(time.Now(), task.Date, task.Repeat)
+				if err != nil {
+					msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"api.NextDate Error"}, false)
+					w.WriteHeader(errInt)
+					w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+					w.Write(msg)
+					return
+				}
+			}
+		case task.Repeat == "" && task.Date != "":
+			dataTime, err := time.Parse("20060102", task.Date)
+			if err != nil {
+				msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"Parse to date error"}, false)
+				w.WriteHeader(errInt)
+				w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+				w.Write(msg)
+				return
+			}
+			if dataTime.Before(time.Now()) {
+				date = time.Now().Format("20060102")
+			} else {
+				date = task.Date
+			}
+		default:
+			date = task.Date
+			if err != nil {
+				msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"api.NextDate Error"}, false)
+				w.WriteHeader(errInt)
+				w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+				w.Write(msg)
+				return
+			}
+		}
+
+		//// добавляем в БД новую задачу
+		lastID, err := serverJob.AddTask(date, task.Title, task.Comment, task.Repeat)
+		if err != nil {
+			msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"dicerelization error"}, true)
+			w.WriteHeader(errInt)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			w.Write(msg)
+			return
+		}
+
+		var answear TaskResponseID = TaskResponseID{lastID}
+
+		answearJSON, err := json.Marshal(&answear)
+		if err != nil {
+			msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"dicerelization error"}, true)
+			w.WriteHeader(errInt)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			w.Write(msg)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(answearJSON)
+
+	}
+}
+
+// GetTasksHundler function to get the nearest tasks. The number of tasks is set by the variable NumberOfOupTasksString. The variable is defined in the .env file
+// this function also handles search by date or text.
+// GetTasksHundler функция для нахождения ближайших задач. Количество задач задается переменной NumberOfOutTasksString. Переменная определена в файле .env
+// в данной функции так же обрабатывается поиск по дате или тексту.
+func GetTasksHundler(serverJob ServerJob, NumberOfOutTasksString string) http.HandlerFunc {
```

Comment:
Обычно если название пакета уже не вставляют в название функции, а то получиться `handlers.GetTasksHandler`
```suggestion
func GetTasks(serverJob ServerJob, NumberOfOutTasksString string) http.HandlerFunc {
```